Lib/asyncore.py

```diff
@@ -131,33 +131,30 @@ def poll(timeout=0.0, map=None):
             is_r = obj.readable()
             is_w = obj.writable()
             if is_r:
-                r.append(fd)
+                r.append(obj)
             # accepting sockets should not be writable
             if is_w and not obj.accepting:
-                w.append(fd)
+                w.append(obj)
             if is_r or is_w:
-                e.append(fd)
+                e.append(obj)
         if [] == r == w == e:
             time.sleep(timeout)
             return
 
         r, w, e = select.select(r, w, e, timeout)
```

Comment:
asyncore is close to its death. I'm not able to accept such large changes. I would prefer changes as small as possible, so please use the same method than for poll():

```
ready = [(map[fd], flags) for fd, flags in r]
```

The thing is that asyncore code base is old and error-prone, the test suite is very small. I cannot affort the risk of introducing a regression.